### ISSUE TITLE
Bad Framerate

### ISSUE THREAD

**Issue by eliteSchwein** [perthcpe23/android-mjpeg-view#9]
**Describe the bug**
I have only 3 FPS or so

**To Reproduce**
Open a HD 30fps Stream

**Expected behavior**
Run at 30fps

**Environment**
 - Device: MiMix
 - OS: 11
 - MjpegView version 1.0.9

**Additional context**

Written in Kotlin but the behavior should be the same in Java

val camViewer: MjpegView = root.findViewById(R.id.webcam_stream)
        camViewer.mode = MjpegView.MODE_FIT_WIDTH
        camViewer.setUrl(printerData.getString("webcamurl"))
        camViewer.isRecycleBitmap = true
        camViewer.startStream()

**Comment by perthcpe23** [comment]
Can you try using this streaming URL [https://server.hafka.co/mjpeg/](url) and confirm that the issue is still stand?

**Comment by eliteSchwein** [comment]
lower res helps a bit, but i have a HD stream or FHD Stream

**Comment by perthcpe23** [comment]
Sorry to ask but is it your internet connection speed? I try mentioned URL above and it works fine, I got 30 fps (judging by eyes).

**Comment by eliteSchwein** [comment]
i have 160mbit, the mjpeg is smooth in browser but not app, i tried it with 640*480 aswell...

**Comment by perthcpe23** [comment]
Is your streaming server publicly accessible? I tried mentioned URL and it works fine on my Sony 5 II. There are too many factors to start discuss one-by-one here. I guess it may have something to do with streaming content your server generates which is supported by browser but, not this library (yet).

**Comment by eliteSchwein** [comment]
i use a local url, with webview it works fine

**Comment by perthcpe23** [comment]
So, a connection speed is less likely an issue here. Can you run this command for a few seconds and send me dump.bin file? I suspect the issue is with stream content.

`curl --output dump.bin "<YOUR-MJPEG-URL>"`

**Comment by eliteSchwein** [comment]
https://files.eliteschw31n.de/Temp/dump.bin

**Comment by eliteSchwein** [comment]
with https://server.hafka.co/mjpeg/ it runs smooth, but again thats low res

**Comment by perthcpe23** [comment]
I see. Let me try FHD version of  [https://server.hafka.co/mjpeg/](https://server.hafka.co/mjpeg/). I have never worked with FHD mjpeg.

**Comment by eliteSchwein** [comment]
i use also 25mbit

**Comment by perthcpe23** [comment]
I changed [https://server.hafka.co/mjpeg/](https://server.hafka.co/mjpeg/) to FHD and fps indeed drops. Now that I can reproduce the issue, I will continue investigating the issue.

**Comment by eliteSchwein** [comment]
Yeah i can short your search, its drawBitmap... I tried it with opengl because of that but i domt understand how i can use opengl, maybe it helps you?

**Comment by perthcpe23** [comment]
Thanks, wow that is a whole new level of rendering.

**Comment by perthcpe23** [comment]
Reduce video quality did help [https://server.hafka.co/mjpeg?type=fhd_low_quality](https://server.hafka.co/mjpeg?type=fhd_low_quality).

**Comment by eliteSchwein** [comment]
On what Bitrate? And also i dont play just a mjpeg file, i have a live stream

**Comment by perthcpe23** [comment]
Yes Bitrate. If you have control over streaming server. I think that is a plausible workaround.

**Comment by eliteSchwein** [comment]
Yeah i have the access over the bitrate, but other people that use my app maybe dont know how to change that on their live stream

**Comment by perthcpe23** [comment]
I think the issue is with how I separate each image from stream using only boundary, instead of using content length and boundary together. The more bytes in each image (higher quality) the more number of time to check for boundary and the longer each checking took. I'm going to try using content length see if it help.

**Comment by tykling** [comment]
Hello. Any news on this?
